internal/policy/policy.go

```diff
@@ -765,3 +770,54 @@ func (s *State) findDelegationEntry(roleName string) (*tuf.Delegation, error) {
 		}
 	}
 }
+
+func ListRules(ctx context.Context, repo *git.Repository) ([]*DelegationDepth, error) {
+	state, err := LoadCurrentState(ctx, repo)
+	if err != nil {
+		return nil, err
+	}
+
+	topLevelTargetsMetadata, err := state.GetTargetsMetadata(TargetsRoleName)
+	if err != nil {
+		return nil, err
+	}
+
+	delegationTargetsMetadata := map[string]*tuf.TargetsMetadata{}
```

Comment:
Any reason to populate these upfront? I suggest using `GetTargetsMetadata` when `HasTargetsRole` is for some delegation. That'll limit loading to _reachable_ roles.